backend/events/apps.py

```diff
@@ -40,53 +40,82 @@ def get_event_loop():
         loop.close()
 
 
-def tuple_get_worker(tuple_type, _tuple):
-    if tuple_type == 'aggregatetuple':
-        return _tuple['worker']
-    return _tuple['dataset']['worker']
+def tuple_get_worker(event_type, _asset):
+    if event_type == 'aggregatetuple':
+        return _asset['worker']
+    return _asset['dataset']['worker']
 
 
-def on_tuples(cc_event, block_number, tx_id, tx_status):
-    payload = json.loads(cc_event['payload'])
+def on_tuples(tx_status, event_type, _asset):
+
     owner = get_owner()
     worker_queue = f"{LEDGER['name']}.worker"
 
-    for tuple_type, _tuples in payload.items():
-        if not _tuples:
+    key = _asset['key']
+    status = _asset['status']
+
+    if tx_status != 'VALID':
+        logger.error(
+            f'Failed transaction on task {key}: type={event_type}'
+            f' status={status} with tx status: {tx_status}')
+        return
```

Comment:
This should not be done for all events?

Reply:
What do you mean ? This is the transaction status, if the transaction is not valid, it's not commited  in the ledger !

Reply:
The transaction status is checked only when processing the tuples event. 

I'm wondering if it should be checked for the compute_plan event as well?

Reply:
It has already be done in the meantime.